index.bs

```diff
@@ -10877,35 +10882,46 @@ defined on that interface.
 These properties are described in more detail
 in sections [[#es-attributes]] and [[#es-operations]].
 
-As with the [=interface object=],
-the [=interface prototype object=] also has properties that
-correspond to the [=constants=] defined on that interface,
-described in [[#es-constants]].
-
-If the [{{NoInterfaceObject}}] extended attribute was not specified on the interface,
-then the [=interface prototype object=] must also
-have a <code class="idl">constructor</code> data property with attributes
-{ \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
-whose value is a reference to the [=interface object=] for the interface.
-
-<div algorithm="value of the [[Prototype]] internal slot">
+<div algorithm="to create an interface prototype object">
 
-    The [=interface prototype object=]
-    for a given interface |A| must have a
-    \[[Prototype]] [=internal slot=] whose value is returned from
-    the following steps:
+    The [=interface prototype object=] for a given [=interface=] |interface| and [=Realm=] |realm|
+    is created as follows:
 
-    1.  If |A| is declared with the [{{Global}}] [=extended attribute=], and
-        |A| [=support named properties|supports named properties=], then
-        return the [=named properties object=]
-        for |A|, as defined in [[#named-properties-object]].
-    1.  Otherwise, if |A| is declared to inherit from another
-        interface, then return the
-        [=interface prototype object=]
-        for the inherited interface.
-    1.  Otherwise, if |A| is declared with the [{{LegacyArrayClass}}]
-        extended attribute, then return {{%ArrayPrototype%}}.
-    1.  Otherwise, return {{%ObjectPrototype%}}.
+    1.  Let |proto| be null.
+    1.  If |interface| is declared with the [{{Global}}] [=extended attribute=],
+        and |interface| [=support named properties|supports named properties=],
+        then set |proto| to the [=named properties object=] of |interface|,
+        as defined in [[#named-properties-object]].
+    1.  Otherwise, if |interface| is declared to inherit from another interface,
+        then set |proto| to the [=interface prototype object=] in |realm|
+        of that [=inherited interface=].
+    1.  Otherwise, if |interface| is declared with the [{{LegacyArrayClass}}] [=extended attribute=],
+        then set |proto| to |realm|.\[[Intrinsics]].[[{{%ArrayPrototype%}}]].
+    1.  Otherwise, set |proto| to |realm|.\[[Intrinsics]].[[{{%ObjectPrototype%}}]].
```

Comment:
Feels like the pattern described in https://infra.spec.whatwg.org/#variables is the issue here. I mean you literally have an if…otherwise if…otherwise set of statements right above it, so there's absolutely no chance for this to happen, and yet it's not super obvious to the eye.

Should we write the below instead?

```
    1.  Let |proto| be |realm|.\[[Intrinsics]].[[{{%ObjectPrototype%}}]]..
    1.  If |interface| is declared with the [{{Global}}] [=extended attribute=],
        and |interface| [=support named properties|supports named properties=],
        then set |proto| to the [=named properties object=] of |interface|,
        as defined in [[#named-properties-object]].
    1.  Otherwise, if |interface| is declared to inherit from another interface,
        then set |proto| to the [=interface prototype object=] in |realm|
        of that [=inherited interface=].
    1.  Otherwise, if |interface| is declared with the [{{LegacyArrayClass}}] [=extended attribute=],
        then set |proto| to |realm|.\[[Intrinsics]].[[{{%ArrayPrototype%}}]].
```

Reply:
OK, so on second thought, and after having tried it out, your solution is better.